drivers/vcnl40x0/include/vcnl40x0_params.h

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (C) 2017 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_vcnl40X0
+ *
+ * @{
+ * @file
+ * @brief       Default configuration for VCNL40X0
+ *
+ * @author      Alexandre Abadie <alexandre.abadie@inria.fr>
+ */
+
+#ifndef VCNL40X0_PARAMS_H
+#define VCNL40X0_PARAMS_H
+
+#include "board.h"
+#include "vcnl40x0.h"
+#include "vcnl40x0_internals.h"
+#include "saul_reg.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @name    Default configuration parameters for the VCNL40X0
+ * @{
+ */
+#ifndef VCNL40X0_PARAM_I2C_DEV
+#define VCNL40X0_PARAM_I2C_DEV         I2C_DEV(0)
+#endif
+#ifndef VCNL40X0_PARAM_I2C_ADDR
+#define VCNL40X0_PARAM_I2C_ADDR        VCNL40X0_ADDR
+#endif
+
+#define VCNL40X0_PARAMS_DEFAULT        { .i2c_dev = VCNL40X0_PARAM_I2C_DEV, \
+                                         .i2c_addr = VCNL40X0_PARAM_I2C_ADDR, \
+                                         .led_current = 2, \
+                                         .proximity_rate = VCNL40X0_PROXIMITY_RATE_31, \
+                                         .ambient_avg = VCNL40X0_AMBIENT_AVERAGE_32, \
+                                         .ambient_rate = VCNL40X0_AMBIENT_RATE_2 }
+/**@}*/
+
+/**
+ * @brief   Configure VCNL40X0
+ */
+static const vcnl40x0_params_t vcnl40x0_params[] =
+{
+#ifdef VCNL40X0_PARAMS_CUSTOM
+    VCNL40X0_PARAMS_BOARD,
+#else
+    VCNL40X0_PARAMS_DEFAULT,
+#endif
+};
+
+/**
+ * @brief   Configure SAUL registry entries
+ */
+static const saul_reg_info_t vcnl40x0_saul_reg_info[][2] =
+{
+    {
+        { .name = "vcnl40x0-proximity" },
```

Comment:
yes, I know - you introduced that syntax. However, I think its redundant and has no benefit, because each sensor is defined by its type and unit already. Further remember: longer string require more memory, too.